coffeecard/CoffeeCard.Models/DataTransferObjects/v2/Purchase/PaymentDetails.cs

```diff
@@ -22,5 +23,20 @@ public abstract class PaymentDetails
         /// <example>f5cb3e0f-3b9b-4f50-8c4f-a7450f300a5c</example>
         [Required]
         public string OrderId { get; set; }
+        
+        /// <summary>
+        /// Id for a payment
+        /// </summary>
+        /// <example>186d2b31-ff25-4414-9fd1-bfe9807fa8b7</example>
+        [Required]
+        public string PaymentId { get; set; }
+
+        /// <summary>
+        /// The status of the payment for the purchase
+        /// </summary>
+        /// <example>PurchaseStatus.PendingPayment</example>
+        [Required]
+        public PurchaseStatus PurchaseStatus { get; set; }
```

Comment:
I do not think this is the correct solution to place the Purchase Status in the PaymentDetails class. We already have the Purchase status as part of the `InitiatePurchaseResponse` class, hence we would now track/show the same state in two places which does not make sense.

Reply:
Good point, changed it so the initiate payment method now returns a tuple with the payment details, and the purchaseStatus